lib/datadog/enable_core_dumps.rb

```diff
@@ -0,0 +1,40 @@
+# This helper is used to enable core dumps for the current Ruby app. This is useful when debugging native-level
+# crashes.
+#
+# It can be enabled simply by adding `require 'datadog/enable_core_dumps'` to start of the app.
+
+lambda do
```

Comment:
I think this could go into `Datadog::Kit` (e.g `Datadog::Kit::CoreDump`, loaded via `require 'datadog/kit/core_dump/enable'`, WDYT?

I'd like to use `Datadog.configure` for that but I guess having a config for that would delay the enablement though, when a core dump may happen before config is effective.

Reply:
> I think this could go into Datadog::Kit (e.g Datadog::Kit::CoreDump, loaded via require 'datadog/kit/core_dump/enable', WDYT?

Hmmm. I still don't understand the semantics of Kit. I'm not sure if fits in there... 

How strongly do you think this should go in there? I need a bit more convincing ;)

> I'd like to use Datadog.configure for that but I guess having a config for that would delay the enablement though, when a core dump may happen before config is effective.

Yeah, it's definitely a trade-off. Since I don't think this would be used that often, I'm leaning towards the `require` version, but don't mind changing it (now, or in the future).

Reply:
My thinking is keeping clear delineation of features, products, and dependencies by leveraging namespaces (and maybe some day, separate gems), therefore I feel we should not drop stuff in `lib/datadog` / `Datadog` top level module:

```
$ ls lib/datadog 
appsec	   ci	  core	   kit	   opentracer	  profiling	tracing
appsec.rb  ci.rb  core.rb  kit.rb  opentracer.rb  profiling.rb	tracing.rb
```

`Datadog::Core` is internal stuff products depend on, `Datadog::Kit` is about features/function not related to a specific product, each feature (or set of features) being encapsulated within its own namespace.

Reply:
I stand convinced! Changed in 274d505ea91aa032edf0a8980e08b25d839ebdc1